src/main/scala/introcourse/level05/ExceptionExercises.scala

```diff
@@ -105,37 +116,59 @@ object ExceptionExercises {
     * Hint: Use `map` and `collect`
     *
     * What issues do you run into (if any)?
+    * How should I handle the .map when there is a Throwable?
+    * In the Java world maybe could do null, and filter nulls, but wartremover complains, so perhaps create Option and filter
     */
-  def createValidPeople: List[Person] = {
-    personStringPairs.map {
-      case (name, age) =>
-        try {
-          ???
-        } catch {
-          case _: EmptyNameException       => ???
-          //handle in any other exception here
-        }
-    }
+  def createValidPeople: List[Person] = personStringPairs.map {
+    case (name, age) =>
+      try {
+        Some(createPerson(name, age))
+      } catch {
+        case _: Throwable => None
+      }
+  }.collect {
+    case Some(person) => person
   }
 
   /**
     * Implement the function collectErrors that collects all the Exceptions
     * that occur while processing personStringPairs. It should not throw any Exceptions.
-    * It should only catch Exceptions thrown by createPerson.
+    * It should only catch Exceptions thrown by createPerson.Ò
     *
     * scala> collectErrors
     * > List(InvalidAgeValueException: provided age is invalid: 5o,
-    *        InvalidAgeRangeException: provided age should be between 1-120: 200,
-    *        InvalidAgeRangeException: provided age should be between 1-120: 0,
-    *        EmptyNameException: provided name is empty)
+    * InvalidAgeRangeException: provided age should be between 1-120: 200,
+    * InvalidAgeRangeException: provided age should be between 1-120: 0,
+    * EmptyNameException: provided name is empty)
     *
     * Hint: Use `map` and `collect`
     *
     * What issues do you run into (if any)?
+    * Needed to catch the exception, and pass it on
+    * Using filtering on instanceOf, required further typecasting using asInstanceOf
     */
   def collectErrors: List[Exception] = {
     personStringPairs.map {
-      case (name, age) => ???
+      case (name, age) =>
+        try {
+          createPerson(name, age)
+        } catch {
+          case exception: Exception => exception
+        }
+    }.collect {
+      case exception: Exception => exception
     }
   }
+
+  //  def collectErrors: List[Exception] = {
+  //    personStringPairs.map {
+  //      case (name, age) =>
+  //        try {
+  //          createPerson(name, age)
+  //        } catch {
+  //          case exception: Exception => exception
+  //        }
+  //    }.filter(_.isInstanceOf[Exception])
```

Comment:
TODO: Find out if i it is ok to check for class types and cast them instead of pattern matching?